### DISCUSSION TITLE
Debian Packages?

### DISCUSSION THREAD

**Discussion by proffalken** [valkey-io/valkey#391] [Q&A] [answered]
Hi,

I see on the downloads page that RPM's are available, are there plans to make .deb packages available too?

**Comment by zuiderkwast** [comment, marked as answer]
Yes, there is an RFP (Request For Packaging) filed here: https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=1068342

@guillemj FYI

**Reply by guillemj** [reply]
BTW this then turned into an ITP (Intent To Package), and is now sitting in the NEW upload queue pending archive admin approval (https://ftp-master.debian.org/new/valkey_7.2.5+dfsg1-1.html and https://ftp-master.debian.org/new/valkey_7.2.5+dfsg1-2.html). Once it's approved it will get into unstable, a new source-only upload will be needed so that it can migrate to testing (binary uploads are not permitted to migrate to force all artifacts to be built from a known environment in the Debian build daemons), and then if no severe bugs are found, and the various QA tests pass, it will migrate to testing in a range from a couple of days to 5 days or so.

**Reply by zuiderkwast** [reply]
@guillemj the list of binaries looks like it's taken from an extremely old redis version. That's not the binaries we provide.

We have valkey-server, valkey-cli and valkey-benchmark, plus some symlinks to valkey-server with other names (valkey-sentinel, valkey-check-aof, etc.)

We have scripts to build man pages too now, in the repo valkey-doc. You can take any version since the docs are not versioned. (They describe all versions and mention when each thing was added.)

**Reply by guillemj** [reply]
I have not really reviewed the packaging, but I'd assume that's what it looked like in relase 7.2.5, and the packaging probably was adapted from the redis one.

Once this gets accepted into the archive, then I'd expect updates to start flowing more often, and for the maintainer to then update to the latest upstream release.

**Comment by proffalken** [comment]
Perfect, thanks.

**Comment by zuiderkwast** [comment]
We can leave it open for others to see, ok? I marked my answer as "accepted".

**Reply by proffalken** [reply]
Sure, sounds good!

**Comment by gesundes** [comment]
Can we have it in packages along with tar.gz?

**Reply by zuiderkwast** [reply]
We could, but IMO it's better that it's packaged by the Debian project, just like the RPMs are prvided by Fedora. The distros provide source and several binary packages for various hardware.

**Reply by zuiderkwast** [reply]
Otoh, if we build our own deb files, we could do it immediately on each new release. @roshkhatri wdyt?

**Reply by guillemj** [reply]
This could be problematic if people then end up switching between packages from Debian and upstream generates ones and vice-versa.

**Comment by maherbeg** [comment]
Is it possible to get this into debian bookworm-backports? That would make transitioning systems into debian-trixie a bit easier too!

**Reply by guillemj** [reply]
Perhaps the maintainers would be interested in preparing a backport, someone might need to ask them. :)

Otherwise some other Debian Developer could prep it, and take responsibility.

**Reply by maherbeg** [reply]
I'm happy to ask! I wasn't sure what the process here is between folks on Valkey and the packaging teams of the various distros. I didn't want to step on any toes or create additional maintenance burdens.

**Comment by bilkoua** [comment]
Unfortunately, I kept waiting for a proper repo with separated versions and up-to-date releases, but it never happened, so I built one myself.
I set up a GitHub Actions CI pipeline that checks for new upstream versions, and when it finds one, it builds the package and publishes it as an APT repository format on **Cloudflare R2**.
Currently, the following targets are supported:
| Distribution | Codename | Architecture |
| ------------ | -------- | ------------ |
| Debian 13 | `trixie` | `amd64`, `arm64` |
| Debian 12 | `bookworm` | `amd64`, `arm64` |
| Ubuntu 24.04 LTS | `noble` | `amd64`, `arm64` |
| Ubuntu 22.04 LTS | `jammy` | `amd64`, `arm64` |

Adding more is straightforward if needed.
What I need now is help with testing.

Valkey repo page: https://pkgs.bil.co.ua/pages/valkey.html
Git repo: [https://github.com/community-pkgs/packages/tree/main/packages/valkey](https://github.com/community-pkgs/packages/tree/main/packages/valkey)

**Comment by roshkhatri** [comment]
Hey, We are working on adding these soon here: https://github.com/valkey-io/valkey-release-automation/pull/47
